pkg/cli/zip_test.go

```diff
@@ -769,6 +771,228 @@ requesting table details for system.web_sessions... writing: debug/schema/system
 requesting table details for system.zones... writing: debug/schema/system/zones.json
 `
 	assert.Equal(t, expected, out)
+
+	// Now mark the stopped node as decommissioned, and check that zip
+	// skips over it.
+	s := tc.Server(0)
+	conn, err := s.RPCContext().GRPCDialNode(s.ServingRPCAddr(), s.NodeID(),
+		rpc.DefaultClass).Connect(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	as := serverpb.NewAdminClient(conn)
+	req := &serverpb.DecommissionRequest{
+		NodeIDs:         []roachpb.NodeID{2},
+		Decommissioning: true,
+	}
+	if _, err := as.Decommission(context.Background(), req); err != nil {
+		t.Fatal(err)
+	}
+	kvserver.TimeUntilStoreDead.Override(&s.ClusterSettings().SV, kvserver.TestTimeUntilStoreDead)
+
+	testutils.SucceedsSoon(t, func() error {
+		out, err = c.RunWithCapture("debug zip " + os.DevNull)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Strip any non-deterministic messages.
+		out = eraseNonDeterministicZipOutput(out)
+
+		const expected2 = `debug zip ` + os.DevNull + `
```

Comment:
Just an idea (i.e. no pressure): this test could benefit from datadriven. Something like

```go
func Test(t *testing.T) {
  tc := _ // ...
  // ...
  decommissionSomething(tc)
  datadriven.Run("testdata/zip_decommissioned_node", func(td *datadriven.TestData) string {
    fmt.Println(eraseNonDeterministicZipOutput(c.RunWitHCapture("...")))
  })
  // do something else...
  _
  // ...
  datadriven.Run("testdata/zip_something_else...", ...)
}
```

The testdata files would just be


````
----
establishing RPC connection to ...
[...]
```